tests/unit/tasks/models/jobsModel.spec.ts

```diff
@@ -0,0 +1,363 @@
+import { syncClientMock, triggerSyncMock } from '../../../mocks/clients/syncClient';
+import { configMock, init as initMockConfig, setValue } from '../../../mocks/config';
+import { linkBuilderMock } from '../../../mocks/linkBuilder';
+import { OperationTypeEnum } from '../../../../src/serviceClients/syncClient';
+import { OperationStatus } from '../../../../src/common/enums';
+import { mergeMock, metadataMergerMock } from '../../../mocks/metadataMerger';
+import { IPublishMapLayerRequest, PublishedMapLayerCacheType } from '../../../../src/layers/interfaces';
+
+let jobsManager: JobsManager;
+
+const jobId = 'c3e8d0c6-6663-49e5-9257-323674161725';
+const taskId = '517059cc-f60b-4542-8a41-fdd163358d74';
+
+describe('JobsManager', () => {
+  const ingestionNewJobType = 'Ingestion_New';
+  const ingestionUpdateJobType = 'Ingestion_Update';
+  const tileSplitTask = 'tilesSplitting';
+  const tileMergeTask = 'tilesMerging';
+  beforeEach(function () {
+    jest.resetAllMocks();
+    setValue('mapServerCacheType', 'FS');
+    initMockConfig();
+    jobsManager = new JobsManager(
+      configMock,
+      jsLogger({ enabled: false }),
+      syncClientMock,
+      jobManagerClientMock,
+      mapPublisherClientMock,
+      catalogClientMock,
+      linkBuilderMock,
+      metadataMergerMock
+    );
+  });
+
+  describe('completeWorkerTask', () => {
+    const testMetadata = {
+      description: 'test desc',
+      productType: ProductType.ORTHOPHOTO_HISTORY,
+      productName: 'test',
+      productVersion: '1',
+      productId: 'test',
+      maxResolutionDeg: 2.68220901489258e-6,
+      tileOutputFormat: TileOutputFormat.JPEG,
+    };
+
+    const mapPublishReq: IPublishMapLayerRequest = {
+      name: `test-${testMetadata.productType}`,
+      tilesPath: `test/${testMetadata.productType}`,
+      cacheType: PublishedMapLayerCacheType.FS,
+      format: TileOutputFormat.JPEG,
+    };
+
+    const catalogReqData = {
+      metadata: { ...testMetadata },
+      links: undefined,
+    };
+
+    it('publish layer to catalog once if all tasks are done for RASTER_MAP', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ mapServerCacheType: 'fs', 'tiling.zoomGroups': '0-10,11,12,13,14,15,16,17,18' });
+      setValue('shouldSync', true);
+      setValue('ingestionNewJobType', ingestionNewJobType);
+      setValue('ingestionTaskType', { tileMergeTask, tileSplitTask });
+
+      const rasterMapTestData = { ...testMetadata };
+      rasterMapTestData.productType = ProductType.RASTER_MAP;
+
+      getJobStatusMock.mockReturnValue({
+        isCompleted: true,
+        isSuccessful: true,
+        metadata: rasterMapTestData,
+        relativePath: `test/${ProductType.RASTER_MAP}`,
+        type: ingestionNewJobType,
+      });
+
+      getTaskMock.mockReturnValue({
+        id: taskId,
+        jobId: jobId,
+        type: tileSplitTask,
+      });
+
+      // tasksManager = new TasksManager(
```

Comment:
why the remarks?

Reply:
should have been deleted, i removed it, old test initialized task manager every test, i made it initialize once before all and its called job manager now instad of task manager , see old tests => https://github.com/MapColonies/discrete-overseer-creator/blob/master/tests/unit/tasks/models/tasksModel.spec.ts